src/mesh/gmsh_reader.cpp

```diff
@@ -706,37 +1108,162 @@ bool get_new_rotated_indices(const dealii::CellAccessor<dim, spacedim>& cell,
+                    if (matching[i] == 1) all_matching = false;
+                }
+
+                //Boolean for good rotation from the previous matching for loop
+                good_rotation = all_matching;
+
+                //If rotation is good, i.e., matching vertices, then we break.
+                if (good_rotation) {
+                    break;
+                }
+
+                //If we did not find a good rotation, we rotate in the Z-axis,
+                //Swap the high_order_vertices_id_rotated by Z-axis rotation
+                high_order_vertices_id = high_order_vertices_id_rotated;
+                for (unsigned int i = 0; i < high_order_vertices_id.size(); ++i) {
+                    high_order_vertices_id_rotated[i] = high_order_vertices_id[rotate_z90degree_3D[i]];
+                }
+            }
+
+            //If rotation is good, i.e., matching vertices, then we break.
+            if (good_rotation) {
+                break;
+            }
+
+            //If we did not find a good rotation, we rotate in the Y-axis,
+            //Swap the high_order_vertices_id_rotated by Y-axis rotation
+            high_order_y_id_rotated = high_order_vertices_id_rotated;
+            for (unsigned int i = 0; i < high_order_vertices_id.size(); ++i) {
+                high_order_vertices_id_rotated[i] = high_order_y_id_rotated[rotate_y90degree_3D[i]];
+            }
+        }
+
+        //If rotation is good, i.e., matching vertices, then we break.
+        if (good_rotation) {
+            break;
+        }
+
+        //If we did not find a good rotation, we rotate in the X-axis
+        //Swap the high_order_vertices_id_rotated by X-axis rotation
+        high_order_x_id_rotated = high_order_vertices_id_rotated;
+        for (unsigned int i = 0; i < high_order_vertices_id.size(); ++i) {
+            high_order_vertices_id_rotated[i] = high_order_x_id_rotated[rotate_x90degree_3D[i]];
+        }
+
+    }
+
+    if (good_rotation) {
+        return true;
+    }
+
+    return false;
+}
+
+
+// template <int dim, int spacedim>
+// std::shared_ptr< HighOrderGrid<dim, double> >
+// read_gmsh(std::string filename, bool periodic_x, bool periodic_y, bool periodic_z, int x_periodic_1, int x_periodic_2, int y_periodic_1, int y_periodic_2, int z_periodic_1, int z_periodic_2, true, int requested_grid_order)
 
 template <int dim, int spacedim>
 std::shared_ptr< HighOrderGrid<dim, double> >
-read_gmsh(std::string filename, int requested_grid_order, const bool use_mesh_smoothing)
+read_gmsh(std::string filename, 
+          const bool periodic_x, const bool periodic_y, const bool periodic_z, 
+          const int x_periodic_1, const int x_periodic_2, 
+          const int y_periodic_1, const int y_periodic_2, 
+          const int z_periodic_1, const int z_periodic_2, 
+          const bool mesh_reader_verbose_output,
+          int requested_grid_order,
+          const bool use_mesh_smoothing)
```

Comment:
will need to pass `do_renumber_dofs` and add if statements before the two occurrences of `dealii::DoFRenumbering::Cuthill_McKee`